kong/tools/grpc.lua

```diff
@@ -65,9 +66,12 @@ end
 
 function _M.new()
   local protoc_instance = protoc.new()
+  local kong_init_file, _, _ = loader.which("kong")
+  local kong_include_dir = kong_init_file:gsub("init%.lua$", "include")
```

Comment:
Making code too overprotective is not my cup of tea, but this worries me. Two concerns:
* The GRPC module shouldn't "know" about kong internal structure (like that our "main luarock uses an init.lua file") 
* If we for some reason decide to rename init.lua to something else (unlikely, but it might happen if we decide to change how we change modules), this will fail with an obscure message.

My first impulse would be moving this logic to a function in `kong.tools.utils.get_include_module` and also write a test for it, so it breaks the tests if we change things. However I believe the grpc module gets required before utils, so this might not be possible. I don't know where else I would put this function. Kong.global or even kong.constants, perhaps? The path shouldn't change once kong starts after all.

Reply:
* The GRPC module needs to know that from the root of the installed luarock (or source code directory), there is a folder named `kong/include` that stores all the proto files to be loaded. I agree that looking for `kong/init.lua` isn't the best approach but there isn't a luarock API right now for "give me the install root of a rock", only "give me the path to a file in a luarock").
* I'm against moving tiny pieces of code that are only used once into utility modules. I think that just obfuscates the code and makes it harder to read. I did add a new comment here to (hopefully) better explain why I am calling the current API. 
* I think if we renamed `kong/init.lua` the tests I wrote would break trying to call `:gsub` on a nil table. I'm open to adding an assert with a better message like `assert(kong_init_file, "Could not find the 'init.lua' file in the root of the kong luarock")`. Thoughts?

Reply:
Let's go with the assert then.
```suggestion
  local kong_include_dir = kong_init_file:gsub("init%.lua$", "include")
  assert(kong_include_dir, "Could not find the 'init.lua' file in the root of the kong luarock")
```